feeds/src/state/ducks/aggregator/fluxOperations.ts

```diff
@@ -0,0 +1,257 @@
+    }
+  }
+}
+
+const fetchOracleAnswersById = (request: any) => {
+  return async (dispatch: any, getState: any) => {
+    try {
+      const currentLogs = getState().aggregator.oracleAnswers || []
+      const logs = await contractInstance.submissionReceivedLogs(request)
+      const withTimestamp = await contractInstance.addBlockTimestampToLogs(logs)
+      const withGasAndTimeStamp = await contractInstance.addGasPriceToLogs(
+        withTimestamp,
+      )
+
+      const uniquePayload = _.uniqBy(
+        [...withGasAndTimeStamp, ...currentLogs],
+        l => {
+          return l.sender
+        },
+      )
+
+      dispatch(actions.setOracleAnswers(uniquePayload))
+    } catch {
+      console.error('Could not fetch oracle answers')
+    }
+  }
+}
+
+const fetchLatestRequestTimestamp = (request: any) => {
+  return async (dispatch: any) => {
+    try {
+      const logs = await contractInstance.newRoundLogs(request)
+      const startedAt = logs.length && logs[logs.length - 1].startedAt
+      dispatch(actions.setLatestRequestTimestamp(startedAt))
+    } catch {
+      console.error('Could not fetch request time')
+    }
+  }
+}
+
+function fetchMinimumAnswers() {
+  return async (dispatch: any) => {
+    try {
+      const payload = await contractInstance.minimumAnswers()
+      dispatch(actions.setMinumumAnswers(payload))
+    } catch {
+      console.error('Could not fetch minimum answers')
+    }
+  }
+}
+
+function fetchAnswerHistory(fromBlock: number) {
+  return async (dispatch: any) => {
+    try {
+      const payload = await contractInstance.answerUpdatedLogs({ fromBlock })
+      const uniquePayload = _.uniqBy(payload, (e: any) => {
+        return e.answerId
+      })
+
+      dispatch(actions.setAnswerHistory(uniquePayload))
+    } catch {
+      console.error('Could not fetch answer history')
+    }
+  }
+}
+
+function initListeners() {
+  return async (dispatch: any, getState: any) => {
+    contractInstance.listenSubmissionReceivedEvent(async (responseLog: any) => {
+      const { minimumAnswers } = getState().aggregator
+      const oracleAnswers = getState().aggregator.oracleAnswers || []
+      const updatedAnswers = oracleAnswers.map((response: any) => {
+        return response.sender === responseLog.sender ? responseLog : response
+      })
+
+      dispatch(actions.setOracleAnswers(updatedAnswers))
+
+      const latestIdAnswers = _.filter(updatedAnswers, {
+        answerId: responseLog.answerId,
+      })
+
+      if (latestIdAnswers.length >= minimumAnswers) {
+        fetchLatestAnswer()(dispatch)
+        fetchLatestAnswerTimestamp()(dispatch)
+      }
+    })
+
+    contractInstance.listenNewRoundEvent(async (responseLog: any) => {
+      await fetchLatestCompletedAnswerId()(dispatch)
+      dispatch(actions.setPendingAnswerId(responseLog.answerId))
+      dispatch(actions.setLatestRequestTimestamp(responseLog.startedAt))
+    })
+  }
+}
+
+const initContract = (config: any) => {
```

Comment:
I think it might be useful to have a paragraph description here, this function does a lot.